packages/bottender/src/router/index.ts

```diff
@@ -0,0 +1,123 @@
+import Context from '../context/Context';
+
+type MatchPattern = string | Array<string> | RegExp;
+
+type RoutePattern = '*' | RoutePredicate;
+
+type RoutePredicate = (context: Context) => boolean | Promise<boolean>;
+
+type Action = (
+  context: Context,
+  props?: Props
+) => void | Action | Promise<Action>;
+
+type Route = {
+  predicate: RoutePredicate;
+  action: Action;
+};
+
+type Props = {
+  next?: Action;
+  [key: string]: any;
+};
+
+function router(routes: Route[]) {
+  return async function Router(context: Context, { next }: Props = {}) {
+    for (const r of routes) {
```

Comment:
prefer `route` of `routes`

Reply:
`route` already taken by an outter function, so...